kube-derive/src/custom_resource.rs

```diff
@@ -417,21 +429,18 @@ fn generate_hasspec(spec_ident: &Ident, root_ident: &Ident) -> TokenStream {
 /// * `status`: The optional name of the `status` struct to use
 /// * `visibility`: Desired visibility of the generated field
 ///
-/// returns: A tuple of (status_field, status_default, status_impl)
-/// * `status_field` is the code to be used for the field in the main struct
-/// * `status_default` is the initialization code to use in a `Default` implementation
-/// * `status_impl` is the implementation code for the `HasStatus` trait
-fn process_status(root_ident: &Ident, status: &Option<String>, visibility: &Visibility) -> (TokenStream, TokenStream, TokenStream) {
+/// returns: A `StatusInformation` struct
+fn process_status(root_ident: &Ident, status: &Option<String>, visibility: &Visibility) -> StatusInformation {
     if let Some(status_name) = &status {
         let ident = format_ident!("{}", status_name);
 
-        let status_field = quote! {
+        let field = quote! {
```

Comment:
Nit: These could be inlined into the object definition.

Reply:
I'm afraid I don't understand what you mean, could you elaborate/rephrase?

Reply:
Instead of:

```rust
let bar = do_stuff();
Foo {
    bar,
}
```

We can do:

```rust
Foo {
    bar: do_stuff(),
}
```

Reply:
Done